src/haz3lcore/statics/Statics.re

```diff
@@ -425,25 +512,49 @@ and uexp_to_info_map =
       union_m([m_pat, m_body]),
     );
   | Let(pat, def, body) =>
-    let (ty_pat, ctx_pat, _m_pat) = upat_to_info_map(~mode=Syn, pat);
-    let def_ctx = extend_let_def_ctx(ctx, pat, ctx_pat, def);
+    let (ty_pat, ctx_pat, _m_pat) =
+      upat_to_info_map(~mode=Syn, ~body_ids=body.ids @ def.ids, pat);
+    let (def_ctx, is_rec) = extend_let_def_ctx(ctx, pat, ctx_pat, def);
```

Comment:
i feel like this approach with is_rec is a little indirect. not sure if it's the best way, but you could just do a comparison ctx == def_ctx and avoid having to alter extend_let_def